dds/idl/marshal_generator.cpp

```diff
@@ -123,6 +127,81 @@ namespace {
     },
   };
 
+  string type_to_default(AST_Type* type, const string& name, bool is_anonymous = false)
+  {
+    string val;
+    AST_Type* actual_type = resolveActualType(type);
+    Classification fld_cls = classify(actual_type);
+    const bool use_cxx11 = be_global->language_mapping() == BE_GlobalData::LANGMAP_CXX11;
+    if ((fld_cls & CL_STRUCTURE) || (fld_cls & CL_UNION)) {
+      val = string("set_default(") + name + ");\n";
+    } else if (fld_cls & CL_ARRAY) {
+      AST_Array* arr = dynamic_cast<AST_Array*>(actual_type);
+      string cxx_elem = scoped(arr->base_type()->name());
+      string temp = name;
+      if (temp.size() > 2 && temp.substr(temp.size() - 2, 2) == "()") {
+        temp.erase(temp.size() - 2);
+      }
+      temp += "_temp";
+      std::replace( temp.begin(), temp.end(), '.', '_');
+      std::replace( temp.begin(), temp.end(), '[', '_');
+      std::replace( temp.begin(), temp.end(), ']', '_');
+      if (use_cxx11) {
+        std::string n = scoped(type->name());
```

Comment:
```suggestion
        string n = scoped(type->name());
```

Reply:
Personally I don't like that the file is doing this, so I would prefer the how it is.